lib/IRGen/IRGenSIL.cpp

```diff
@@ -6138,6 +6138,11 @@ void IRGenSILFunction::visitCheckedCastAddrBranchInst(
 }
 
 void IRGenSILFunction::visitHopToExecutorInst(HopToExecutorInst *i) {
+  if (!i->getFunction()->isAsync()) {
+    // This should never occur.
+    assert(false && "The hop_to_executor should have been eliminated");
```

Comment:
I mean, it can occur if we inlined a reasync autoclosure... or are you suggesting we handle it differently?

Reply:
I don’t know what the scenario is that you are referring to. I had assumed that the only case where we can create a hop_to_executor inside of an function that is not async was via inlining a noasync apply.

Is there another case I need to handle?

Reply:
This assert is meant to test that earlier inlining has removed any hop_to_executor. But in release builds we just ignore it if it was somehow created differently instead of crashing.

Reply:
This PR also includes a change to the inliner that removes the hop_to_executor.

Reply:
From what I can tell reasync == `noasync` at the SIL level. So we should be good?

Reply:
Oh I see, yeah, we're good! Thanks!